### PR TITLE
Refresh Bug on Uploader Modal 

### PR DESCRIPTION
**Pull request by chloealee**
#### What's this PR do?
- Adds `updateQuantityOrWhyState` function so when we update the quantity or why participated when uploading a post on behalf of the user, the state on the signup page also updates.
  - This function is needed because we only need to update the state, not make a new call since the submit button in the modal already makes that API call. 
- This fixes the bug that we have right now of when you upload a post with a quantity and/or why, the signup page doesn't reflect this information until the page is refreshed. 

#### How should this be reviewed?
- Go to a user's signup page `/signups/:signup_id`
- Upload a photo with a quantity and/or why participated
- You should see these reflected at the top of the signup page when you close the modal without refreshing the page
- No additional events should be created in the events table

#### Relevant tickets
Fixes this [Pivotal Card](https://www.pivotaltracker.com/n/projects/2019429/stories/152292775).

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love.

### CODE REVIEW

File: resources/assets/components/Signup/index.js [hunk 1]
```diff
@@ -217,6 +217,18 @@ class Signup extends React.Component {
     this.hideHistory();
   }
 
+  // Set newstate with updated quantity and why_participated.
+  updateQuantityOrWhyState(quantity, why) {
```
Comment (sbsmith86) [inline resources/assets/components/Signup/index.js]: it is possible for this to just be `updateSignup(signup)` that just takes in a whole signup object and updates the state with that?

File: resources/assets/components/ReportbackUploader/index.js [hunk 2]
```diff
@@ -74,6 +74,8 @@ class ReportbackUploader extends React.Component {
     this.setState({
       media: this.defaultMediaState,
     });
+
+    this.props.updateQuantityOrWhyState(reportback.impact ? reportback.impact : null, reportback.whyParticipated ? reportback.whyParticipated : null);
```
Comment (sbsmith86) [inline resources/assets/components/ReportbackUploader/index.js]: Oh not really needed for this PR but this `ReportbackUploader` needs to be renamed to `PostUploader` and should be refactored to just use rogue terminology.

File: resources/assets/components/Signup/index.js (hunk 1 above)
Comment (chloealee) [reply resources/assets/components/Signup/index.js]: oooo I like that idea! Just updated that!

File: resources/assets/components/ReportbackUploader/index.js (hunk 2 above)
Comment (chloealee) [reply resources/assets/components/ReportbackUploader/index.js]: ok! I made a card for that here: https://www.pivotaltracker.com/n/projects/2019429/stories/152350532

**Pull request merged**